python/cudf/bindings/copying.pyx

```diff
@@ -71,10 +71,17 @@ def apply_gather(in_cols, maps, out_cols=None):
 
      * returns out_cols
     """
+    if in_cols[0].dtype == np.dtype("object"):
+        in_size = in_cols[0].data.size()
+    else:
+        in_size = in_cols[0].data.size
+
     maps = astype(maps, 'int32')
+    maps = modulo(maps, in_size)
```

Comment:
We have a libcudf binaryop implementation for modulo that we should use instead of the numba one.

Reply:
Will you need the new pymod function we will be adding? #1916

Reply:
In this case no, because both inputs are guaranteed to be positive integers.

Reply:
@kkraus14 `maps` can be negative, no? This is why I'm using the numba modulo and unable to use the current libcudf version.

Reply:
It depends on if libcudf is handling negative values in the gather map or if we're normalizing the values first before passing them to libcudf.

Reply:
We're normalizing them here with the call to `moduloo` before passing them to libcudf (it doesn't handle negative indices).

Reply:
The call to `modulo` here invokes a numba kernel instead of using the libcudf binaryop.

Reply:
To be clear: the `pymod` will be needed here in the Cython to convert the gather map into all positive indices before feeding into `gather`. Until it's available, we could use the numba version

Reply:
Add a comment of a TODO: replace with libcudf pymod when available and I'll resolve this 😄

Reply:
Done

Reply:
`pymod` is here. #1985